plugins/module_utils/network/sonic/config/sflow/sflow.py

```diff
@@ -0,0 +1,500 @@
+        """ Execute the module
+
+        :rtype: A dictionary
+        :returns: The result from module execution
+        """
+        result = {'changed': False}
+        warnings = list()
+
+        existing_sflow_facts = self.get_sflow_facts()
+        commands, requests = self.set_config(existing_sflow_facts)
+        if commands and len(requests) > 0:
+            if not self._module.check_mode:
+                try:
+                    edit_config(self._module, to_request(self._module, requests))
+                except ConnectionError as exc:
+                    self._module.fail_json(msg=str(exc), code=exc.errno)
+            result['changed'] = True
+        result['commands'] = commands
+        changed_sflow_facts = self.get_sflow_facts()
+
+        result['before'] = existing_sflow_facts
+        if result['changed']:
+            result['after'] = changed_sflow_facts
+
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_sflow_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = self._module.params['config']
+        have = existing_sflow_facts
+
+        resp = self.set_state(want, have)
+        return to_list(resp)
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A tuple
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration, and REST requests that do it
+        """
+        commands = []
+        requests = []
+        want = remove_none(want)
+        state = self._module.params['state']
+        if state == 'deleted':
+            commands, requests = self._state_deleted(want, have)
+        elif state == 'merged':
+            commands, requests = self._state_merged(want, have)
+        elif state == 'overridden':
+            commands, requests = self._state_overridden(want, have)
+        else:
+            commands, requests = self._state_replaced(want, have)
+        return commands, requests
+
+    def _state_replaced(self, want, have):
+        """ The command generator when state is replaced
+
+        :rtype: A tuple of lists
+        :returns: A list of what commands and state necessary to migrate the current configuration
+                  to the desired configuration, and a list of requests needed to make changes
+        """
+        commands = []
+        requests = []
+
+        if len(want) == 0:
+            # not wanting to replace anything we can just quit
+            return commands, requests
+
```

Comment:
If "validate_sflow_args" is invoked for "merged" state, it should also be invoked here.

Reply:
moved validate to run for all states